The throw keyword should raise an exception if used to exit a thread
